src/social_wallet_api/handler.clj

```diff
@@ -184,126 +195,146 @@
                                                       :currency (:connection query)))))))
              blockchain-transactions)))
 
+(defn wrap-auth [handler]
+  (fn [request]
+    (if @client
+      (if (= (or (-> request :headers (get "X-API-Key"))
+                 (-> request :headers (get "x-api-key")))
+             (get @apikey @client))
+        (handler request)
+        (unauthorized {:error "Could not access the Social Wallet API"}))
```

Comment:
Maybe indicate "wrong API key" as a reason here.
In future we may also want to throttle failures at this point just like login attempts.